dali/kernels/imgproc/resample/resampling_internal_test.cu

```diff
@@ -0,0 +1,470 @@
+
+  int x0 = blockIdx.x * in_w / gridDim.x;
+  int x1 = (blockIdx.x + 1) * in_w / gridDim.x;
+  int y0 = blockIdx.y * out_h / gridDim.y;
+  int y1 = (blockIdx.y + 1) * out_h / gridDim.y;
+  ResampleVert(
+    x0, x1, y0, y1, 0, scale,
+    out, out_stride, in, in_stride, in_h,
+    channels, filter, support);
+}
+
+TEST(Resample, HorizontalGaussian) {
+  auto cv_img = testing::data::image("imgproc_test/checkerboard.png");
+  auto cv_ref = testing::data::image("imgproc_test/ref_out/resample_horz.png");
+  ASSERT_FALSE(cv_img.empty()) << "Couldn't load the image";
+  TensorView<StorageCPU, const uint8_t, 3> img;
+  ASSERT_NO_FATAL_FAILURE((img = view_as_tensor<uint8_t>(cv_img)));
+
+  int channels = img.shape[2];
+  ASSERT_EQ(channels, 3);
+  int H = img.shape[0];
+  int W = img.shape[1];
+  int outW = W / 2;
+  auto gpu_mem_in = memory::alloc_unique<uint8_t>(AllocType::GPU, img.num_elements());
+  auto gpu_mem_out = memory::alloc_unique<uint8_t>(AllocType::GPU, H * outW * channels);
+
+  TensorView<StorageGPU, uint8_t, 3> img_in, img_out;
+  img_in = { gpu_mem_in.get(), img.shape };
+  img_out = { gpu_mem_out.get(), { H, outW, channels } };
+
+  copy(img_in, img);  // NOLINT
+
+  auto filters = GetResamplingFilters(0);
+  ResamplingFilter filter = (*filters)[1];
+
+  int radius = 40;
+  filter.rescale(2*radius+1);
+
+  for (int i = 0; i < 100; i++) {
+    ResampleHorzTestKernel<<<1, dim3(32, 24), ResampleSharedMemSize>>>(
+      img_out.data, outW*channels, outW, img_in.data, W*channels, W, H, channels,
+      filter, filter.support());
+    cudaDeviceSynchronize();
+  }
+
+  cv::Mat out;
+  out.create(H, outW, CV_8UC3);
+  auto img_out_cpu = view_as_tensor<uint8_t, 3>(out);
+  auto img_ref_cpu = view_as_tensor<uint8_t, 3>(cv_ref);
+  copy(img_out_cpu, img_out);  // NOLINT
+  cudaDeviceSynchronize();
+  EXPECT_NO_FATAL_FAILURE(Check(img_out_cpu, img_ref_cpu, EqualEps(1))) <<
+  [&]() {
+    cv::Mat diff;
+    cv::absdiff(out, cv_ref, diff);
+    cv::imwrite("resample_horz_dif.png", diff);
+    return "Test failed. Absolute difference image saved to resample_horz_dif.png";
+  }();
+}
+
+TEST(Resample, VerticalGaussian) {
+  auto cv_img = testing::data::image("imgproc_test/checkerboard.png");
+  auto cv_ref = testing::data::image("imgproc_test/ref_out/resample_vert.png");
+  ASSERT_FALSE(cv_img.empty()) << "Couldn't load the image";
+  TensorView<StorageCPU, const uint8_t, 3> img;
+  ASSERT_NO_FATAL_FAILURE((img = view_as_tensor<uint8_t>(cv_img)));
+
+  int channels = img.shape[2];
+  ASSERT_EQ(channels, 3);
+  int H = img.shape[0];
+  int W = img.shape[1];
+  int outH = H / 2;
+  auto gpu_mem_in = memory::alloc_unique<uint8_t>(AllocType::GPU, img.num_elements());
+  auto gpu_mem_out = memory::alloc_unique<uint8_t>(AllocType::GPU, outH * W * channels);
+
+  TensorView<StorageGPU, uint8_t, 3> img_in, img_out;
+  img_in = { gpu_mem_in.get(), img.shape };
+  img_out = { gpu_mem_out.get(), { outH, W, channels } };
+
+  copy(img_in, img);  // NOLINT
```

Comment:
why the NOLINT?

Reply:
cpplint had a problem with `copy` function supposedly requiring `#include <algorithm>`. cpplint has been modified.